src/layouts/Base.astro

```diff
@@ -45,15 +45,16 @@ const socials: Social[] | undefined = siteConfig.socials?.map((social) => {
     name: social.name || "",
     iconImageID: social.icon?.asset?._id || "",
     link: social.link
-      ? getHrefFromLinkObject(social.link, language, preview)
+      ? getHrefFromLinkObject(social.link, getLocale(), preview)
       : "",
-    alt: social.icon?.asset?.altTexts?.[language] || "",
+    alt: social.icon?.asset?.altTexts?.[getLocale()] || "",
   };
 });
 ---
 
-<html class="overflow-x-clip" lang={language.toLowerCase()}>
+<html class="overflow-x-clip" lang={getLocale().toLowerCase()}>
   <head>
+    <I18nClient />
```

Comment:
Also worth adding/looking into is the `<I18nHead>` component for better SEO: https://astro-i18n.netlify.app/reference/components/i18n-head/

It adds this to the head of pages:
![image](https://github.com/user-attachments/assets/a2107370-8d2d-4f85-9178-fc1c64f726c7)

Which improves SEO a bit.

This doesn't seem to be very configurable so I can see some issues with translated slugs here... I'll make a backlog ticket for it since it isn't that high prio either.